contracts/interchain-token-service/src/contract/execute/mod.rs

```diff
@@ -457,11 +499,200 @@ mod tests {
     const GOVERNANCE: &str = "governance";
     const AXELARNET_GATEWAY: &str = "axelarnet-gateway";
 
-    fn its_address() -> nonempty::HexBinary {
-        HexBinary::from_hex(ITS_ADDRESS)
-            .unwrap()
-            .try_into()
-            .unwrap()
+    #[test]
+    fn should_be_able_to_transfer() {
+        let mut deps = mock_dependencies();
+        init(&mut deps);
+
+        assert_ok!(do_deploy(deps.as_mut(), ethereum(), solana(), token_id()));
+
+        assert_ok!(do_transfer(
+            deps.as_mut(),
+            ethereum(),
+            solana(),
+            token_id(),
+            Uint256::one().try_into().unwrap()
+        ));
+    }
+
+    #[test]
+    fn should_not_be_able_to_transfer_more_than_supply() {
+        let mut deps = mock_dependencies();
+        init(&mut deps);
+
+        assert_ok!(do_deploy(deps.as_mut(), ethereum(), solana(), token_id()));
+
+        assert_err_contains!(
+            do_transfer(
+                deps.as_mut(),
+                solana(),
+                ethereum(),
+                token_id(),
+                Uint256::one().try_into().unwrap()
+            ),
+            Error,
+            Error::TokenSupplyInvariantViolated { .. }
+        );
+    }
```

Comment:
looking at these tests I have no idea why the first one should succeed and the second one shouldn't

Reply:
added some comments and generally tried to clean these up, lmk what you think